Kotlin/app/src/main/java/com/taboola/kotlin/examples/screens/web/WebComposeWidgetFragment.kt

```diff
@@ -42,11 +50,9 @@ fun CustomerWebView(placementInfo: PlacementInfo.Properties) {
                 ViewGroup.LayoutParams.MATCH_PARENT
             )
 
-            //Create taboolaWebWrapperViewModel instance
-            //Inside the WebView apply scope we will add Taboola code to use web integration
-            val taboolaWebWrapperViewModel = TaboolaWebWrapperViewModel()
+
             taboolaWebWrapperViewModel.setupWebViewForTaboola(this)
```

Comment:
@voltfmanSlava  - Why is the setup for the viewModel happening inside the WebView composable?

Reply:
@tomer-br-taboola because i need webview to be passed to taboolaWebWrapperViewModel.

Reply:
@voltfmanSlava  - This seems fundamentally wrong and I'm sure there is a proper way to fix this.